src/core/Header/HeaderButton.tsx

```diff
@@ -78,11 +89,17 @@ export const HeaderButton = (props: HeaderButtonProps) => {
     ...rest,
   };
 
-  return menuItems ? (
-    <DropdownButton {...buttonProps} menuItems={menuItems} />
-  ) : (
-    <Button {...buttonProps} />
-  );
+  if (menuItems && onClick) {
```

Comment:
Previously you had `isSplitButton` variable, it seems you could bring it back and reuse in several places.